src/app/client/src/app/modules/shared-feature/components/onboarding-user-selection/onboarding-user-selection.component.spec.ts

```diff
@@ -0,0 +1,144 @@
+  const mockTelemetryService: Partial<TelemetryService> = {};
+  const mockActivatedRoute: Partial<ActivatedRoute> = {
+    queryParams: of({})
+  };
+  const mockFormService: Partial<FormService> = {
+    getFormConfig: jest.fn().mockReturnValue(of(true)) as any
+  };
+  const mockProfileService: Partial<ProfileService> = {};
+  const mockUserService: Partial<UserService> = {};
+  const mockToasterService: Partial<ToasterService> = {};
+
+  beforeAll(() => {
+    onboardingUserSelectionComponent = new OnboardingUserSelectionComponent(
+      mockResourceService as ResourceService,
+      mockTenantService as TenantService,
+      mockRouter as Router,
+      mockNavigationHelperService as NavigationHelperService,
+      mockTelemetryService as TelemetryService,
+      mockActivatedRoute as ActivatedRoute,
+      mockFormService as FormService,
+      mockProfileService as ProfileService,
+      mockUserService as UserService,
+      mockToasterService as ToasterService
+    );
+  });
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should  create an instance of onboardingUserSelectionComponent', () => {
+    expect(onboardingUserSelectionComponent).toBeTruthy();
+  });
+
+  describe('setup the onboardingUserSelection component', () => {
+    describe('setup ngOnInit function', () => {
+      it('should call setPopupInteractEdata method', () => {
+        //arrange
+        const setPopupInteractEdataSpy = jest.spyOn(onboardingUserSelectionComponent,'setPopupInteractEdata');
+        //act
+        onboardingUserSelectionComponent.ngOnInit();
+        //assert
+        expect(setPopupInteractEdataSpy).toBeCalled();
+      });
+
+      it('should call initialize method',() => {
+        //arrange
+        //@ts-ignore
+        const initializeSpy = jest.spyOn(onboardingUserSelectionComponent,'initialize');
+        //act
+        onboardingUserSelectionComponent.ngOnInit();
+        //assert
+        expect(initializeSpy).toBeCalled();
+      });
+    });
+  });
+
+  describe('setPopupInteractEdata method',() => {
+    const responseData = {
+      id: 'user-type-select',
+      type: 'click',
+      pageid: 'user',
+    };
+
+    it('should set userSelectionInteractEdata property to the responseData', () => {
+      //act
+      onboardingUserSelectionComponent.setPopupInteractEdata();
+      //assert
+      expect(onboardingUserSelectionComponent.userSelectionInteractEdata).toEqual(responseData);
+    });
+  });
+
+  describe('initialize method',() => {
+    it('should call getFormConfig method of the onboard user selection component', () => {
+      //arrange
+      //@ts-ignore
+      const getFormConfigSpy = jest.spyOn( onboardingUserSelectionComponent,'getFormConfig');
+      //act
+      onboardingUserSelectionComponent['initialize']();
+      //assert
+      expect(getFormConfigSpy).toBeCalled();
+    });
+
+    it('should call updateUserSelection', () => {
+      //arrange
```

Comment:
- updateUserSelection method is using profile service and calling updateProfile
- mock or spy is missing for updateProfile